### PR TITLE
releases/describe_firmware: remove bitbox02 dependency

### PR DESCRIPTION
**Pull request by benma**
As per the releases/README, the file is supposed to be easy to
audit. Pulling in a dependency makes audit more difficult. It also
makes the script harder to use, if the user does not have the
dependency installed.

This commit inlines the relevant parse_signed_firmware() function from
bootloader.py.

### CODE REVIEW

**Comment by x1ddos** [comment]
something funny happens with parsing a firmware file but i haven't dug into why:

```
$ describe_firmware.py firmware-btc.v9.6.0.signed.bin
The following information assumes the provided binary was signed correctly; the signatures are not being verified.
This is a Bitcoin-only edition firmware.
The hash of the unsigned firmware binary is (compare with reproducible build):
18a7c2df2ff8521dc629f9751f4404bfaa0ec6b89560b03363ffd8a344efc333
The monotonic firmware version is: 1064304200
The hash of the firmware as verified/shown by the bootloader is:
341c0a0774b18b16426678c19004ef4267726992ad6d7090b6a9ea38afc250d4
```

one sign something's weird is the "monotonic firmware version" field.

File: releases/describe_signed_firmware.py [hunk 1]
```diff
@@ -14,46 +14,53 @@
...
 
-    try:
-        magic, sigdata, firmware = parse_signed_firmware(binary)
-    except ValueError as exception:
-        print(exception)
+    # Split signed firmware into sigdata and firmware
+    if len(signed_firmware) < SIGDATA_LEN:
+        print("firmware too small")
         return 1
+    sigdata, firmware = signed_firmware[:SIGDATA_LEN], signed_firmware[SIGDATA_LEN:]
```
[the remark this answers is not in the document]
Comment (x1ddos) [reply releases/describe_signed_firmware.py:53]: with a patch like this, the hash matches what's in assertion files:

```patch
diff --git releases/describe_signed_firmware.py releases/describe_signed_firmware.py
index f64b7e34..ae899777 100755
--- releases/describe_signed_firmware.py
+++ releases/describe_signed_firmware.py
@@ -50,7 +50,8 @@ def main() -> int:
     if len(signed_firmware) < SIGDATA_LEN:
         print("firmware too small")
         return 1
-    sigdata, firmware = signed_firmware[:SIGDATA_LEN], signed_firmware[SIGDATA_LEN:]
+    magic, firmware = signed_firmware[:MAGIC_LEN], signed_firmware[MAGIC_LEN:]
+    sigdata, firmware = firmware[:SIGDATA_LEN], firmware[SIGDATA_LEN:]

     print(
         "The following information assumes the provided binary was signed correctly; "
```

[the remark this answers is not in the document]
Comment (x1ddos) [reply releases/describe_signed_firmware.py:53]: would be very nice if the signed firmware format was recorded/described somewhere in a readme.

[the remark this answers is not in the document]
Comment (benma) [reply releases/describe_signed_firmware.py]: thanks for testing :O

Fixed now, please check again.

**Review by x1ddos — APPROVED** [review]
LGTM

**Pull request merged**